`mut` closure support (`cannot borrow \`fun\` as mutable, as it is not declared as mutable`)

I'm trying to write errors to a vec in the replacer function:

```rs
let mut errors = Vec::new();
let mut content = lazy_regex::regex_replace_all!(
    "(?:from \"(\\..*)\"|import \"(\\..*)\")",
    &content,
    |whole: &str, from_request: &str, import_request: &str| {
        match replacer(from_request, import_request) {
            Ok(replacement) => replacement,
            Err(e) => {
                errors.push(e);
                whole.to_string()
            }
        }
    },
)
.to_string();
```

but this unfortunately leads to an error:
```rs
error[E0596]: cannot borrow `fun` as mutable, as it is not declared as mutable
{...}
404 | |                     errors.push(e);
    | |                     ------ calling `fun` requires mutable binding due to mutable borrow of `errors`

```

is there any way to allow mut closures inside the macros?

--- a/src/proc_macros/mod.rs
+++ b/src/proc_macros/mod.rs
@@ -178,7 +178,7 @@ fn replacen(input: TokenStream, limit: usize) -> TokenStream {
                     #value,
                     #limit,
                     |caps: &lazy_regex::Captures<'_>| {
-                        let fun = #fun;
+                        let mut fun = #fun;
                         fun(
                             #(#groups),*
                         )

--- a/tests/replace.rs
+++ b/tests/replace.rs
@@ -1,0 +1,18 @@
+
+/// check replacement with a mut closure
+/// See https://github.com/Canop/lazy-regex/issues/27
+#[test]
+fn replace_with_mut_closure() {
+    let input = "5+183/32";
+    let mut last_digits: Vec<u8> = Vec::new();
+    let output = lazy_regex::regex_replace_all!(
+        r"\d*(\d)",
+        input,
+        |number, last_digit: &str| {
+            last_digits.push(last_digit.parse().unwrap());
+            format!("[{number}]")
+        }
+    );
+    assert_eq!(output, "[5]+[183]/[32]");
+    assert_eq!(last_digits, [5, 3, 2]);
+}

